src/os/bluestore/BlueFS.cc

```diff
@@ -1239,17 +1270,23 @@ void BlueFS::_fsync(FileWriter *h)
 {
   dout(10) << __func__ << " " << h << " " << h->file->fnode << dendl;
   _flush(h, true);
-  _flush_wait(h);
-  if (h->file->dirty) {
-    dout(20) << __func__ << " file metadata is dirty, flushing log on "
-	     << h->file->fnode << dendl;
-    _flush_log();
-    assert(!h->file->dirty);
+  uint64_t old_dirty_seq = h->file->dirty_seq;
+  lock.unlock();
+  wait_for_aio(h);
```

Comment:
Yep.  Updated this to only allow one concurrent log flush at a time.  It now passes your first test, but I'm unsure if we want to do something more complex here or not...